README.md
# Pipetography-HPCCode
 A repo for the basic scripts to run pipetography on an HPC cluster (SGE or SLURM).


Root_BIDS_files/CHANGES
Revision history for your dataset

1.0.0 2021-08-06
 - Initialized study directory


streamline_job.sh
#!/bin/bash

# # Wynton Submissions:
#$ -S /bin/bash
#$ -N PP_Streams
#$ -cwd
#$ -q gpu.q
#$ -j y
#$ -l mem_free=64G
#$ -l h_rt=1:00:00 #gross overestimate, probably closer to 5-10min
#$ -o /wynton/protected/home/rad-brain/bsipes/outlog_general

# # Radiology SCS Submissions:
# #SBATCH --job-name=PP_Streams
# #SBATCH --nodes=1
# #SBATCH --ntasks=1
# #SBATCH --nodelist=cronus #I've had best luck with cronus, although other nodes may work fine, too.
# #SBATCH -N 1 --partition=gpu --ntasks-per-node=1
# #SBATCH --gres=gpu:1
# #SBATCH --mem=64G
# #SBATCH --time=1:00:00 #gross overestimate, probably closer to 5-10min
# #SBATCH --output=/home/%u/slurm/%x-%j.log

# Run NVIDIA GPU Streamline generation:
# You will need to request GPU resources on UCSF radiology, for GPU resources with an interactive shell: qrsh -l gpu=# -q rad_gpu.q@host.radiology.ucsf.edu
# Check which GPUs are assigned to you after getting into the qrsh session with `echo $CUDA_VISIBLE_DEVICES`

echo "Running Tractography for sub-${1} ses-${2}"

# Default parameters for the streamline generation code:
NGPUS=1 # number of GPUS (same as what you requested)
MIN_SIGNAL=1.0e-5
CHUNK_SIZE=90000 # GPU paralleled file writing size
ANGLE_DEG=60 # search angle
MAX_ANGLE=1.04719753 # ANGLE_DEG*pi/180
THRESHOLD=0.1
STEP_SIZE=0.5
SEED_DENS=1

bids_root_dir=/wynton/protected/home/rad-brain/bsipes/datasets/BIDS_dir
# bids_root_dir=`pwd`/..

# # From Wynton:
gpustreamlines_path=/wynton/protected/home/rad-brain/bsipes/singularity/gpustreamlines.sif
# gpustreamlines_path=/protected/data/rajlab1/shared_data/singularity/images/gpustreamlines.sif
# # From Radiology SCS:
# gpustreamlines_path=/data/i2/software/singularity/gpustreamlines.sif


#Make Directories for Outputs
mkdir -p $bids_root_dir/derivatives/pipetography/sub-$1/ses-$2/tracktography
mkdir -p $bids_root_dir/derivatives/streamlines/sub-$1/ses-$2

# Inputs:
#     - BVEC={PATH TO}.bvec
#       BVAL={PATH TO}.bval
#       DTI={PATH TO}.nii.gz
#       MASK=Streamline possible locations mask - whole brain mask
#       ROI=Seed start mask - grey matter white matter interface
BVEC='/input/dwi_space-acpc_res-1mm.bvecs'
BVAL='/input/dwi_space-acpc_res-1mm.bvals'
DTI='/input/dwi_space-acpc_res-1mm.nii.gz'
MASK='/input/dwi_space-acpc_res-1mm_seg-brain_mask.nii.gz'
#ROI='/input/T1w_space-acpc_seg-gmwmi_mask.nii.gz' #Removed for whole-brain seeding
PREFIX='/opt/exec/output/tmp-sub01'  # prefix to name temporary chunk files
TMPFILES='tmp-sub01*' # temp output file with wild card ex: tmp-*

# 2. Execute singularity container:
singularity exec --nv -B $bids_root_dir/derivatives/pipetography/sub-$1/ses-$2/tracktography:/opt/exec/output -B $bids_root_dir/derivatives/pipetography/sub-$1/ses-$2/preprocessed:/input $gpustreamlines_path python /opt/exec/run_dipy_gpu.py $DTI $BVAL $BVEC $MASK --roi_nifti $MASK --chunk-size $CHUNK_SIZE --output-prefix $PREFIX --ngpus $NGPUS --use-fast-write --max-angle $MAX_ANGLE --min-signal $MIN_SIGNAL --tc-threshold $THRESHOLD --step-size $STEP_SIZE --sampling-density $SEED_DENS

echo "Merging track chunks..."
# 3. Merge chunks back together to 1 trk file
singularity exec --nv --pwd /opt/exec/output -B $bids_root_dir/derivatives/pipetography/sub-$1/ses-$2/tracktography:/opt/exec/output $gpustreamlines_path /opt/exec/merge_trk.sh $TMPFILES -o "sub-$1_ses-$2_gmwmi2wm.trk" 

echo "Deleting track chuncks..."
# 5. Delete all tmp chunks 
cd $bids_root_dir/derivatives/pipetography/sub-$1/ses-$2/tracktography 
rm $TMPFILES

echo "Converting TRK file to TCK..."
# 6. Convert trk to tck for mrtrix3 compatibility
# From Wynton:
$bids_root_dir/code/trk2tck -i $bids_root_dir/derivatives/pipetography/sub-$1/ses-$2/tracktography/"sub-$1_ses-$2_gmwmi2wm.trk" -o $bids_root_dir/derivatives/pipetography/sub-$1/ses-$2/tracktography/"sub-$1_ses-$2_gmwmi2wm.tck" -c 3

mv $bids_root_dir/derivatives/pipetography/sub-$1/ses-$2/tracktography/"sub-$1_ses-$2_gmwmi2wm.tck" $bids_root_dir/derivatives/streamlines/sub-$1/ses-$2
rm -r $bids_root_dir/derivatives/pipetography/sub-$1/ses-$2/tracktography

echo "Streamline generation script for sub-${1} ses-${2} complete."

# # Job Summary Wynton:
qstat -j $JOB_ID

# # Job Summary Radiology SCS:
# squeue -j $SLURM_JOB_ID -o "%.18i %.9P %.8j %.8c %.8u %.8T %.10M %.9l %.6D %R %D"

pipetography_all_HPC.sh
#!/bin/bash

# # Wynton Submissions:
#$ -S /bin/bash
#$ -N PP_All
#$ -cwd
#$ -q gpu.q
#$ -j y
#$ -l mem_free=64G
#$ -l h_rt=24:00:00 #gross overestimate, probably closer to 5-10 hours
#$ -o /wynton/protected/home/rad-brain/bsipes/outlog_general

# # Radiology SCS Submissions:
# #SBATCH --job-name=PP_All
# #SBATCH --nodes=1
# #SBATCH --ntasks=1
# #SBATCH --nodelist=cronus #I've had best luck with cronus, although other nodes may work fine, too.
# #SBATCH -N 1 --partition=gpu --ntasks-per-node=1
# #SBATCH --gres=gpu:1
# #SBATCH --mem=64G
# #SBATCH --time=24:00:00 #gross overestimate, probably closer to 5-10 hours
# #SBATCH --output=/home/%u/slurm/%x-%j.log

/bin/echo Running on host: `hostname`.
/bin/echo In directory: `pwd`
/bin/echo Starting on: `date`


bids_root_dir=`pwd`/..

# # From Wynton:
pipetography_path=/protected/data/rajlab1/shared_data/singularity/images/pipetography_latest.sif
# # From Radiology SCS:
# pipetography_path=/data/i2/software/singularity/pipetography_latest.sif


# # # # # # # # # # Preprocessing Job # # # # # # # # # # # # # # # # # # # 

echo "Initiating singularity for preprocesing job..."

singularity exec -e -B $bids_root_dir:/BIDS_dir --pwd $PWD $pipetography_path python /BIDS_dir/code/preprocess_def_singlesub.py $1 $2

date
echo "Preprocessing Job script for sub-{$1} complete."

# # # # # # # # # # Streamlines Job # # # # # # # # # # # # # # # # # # # 


echo "Running Tractography for sub-${1} ses-${2}"

# Default parameters for the streamline generation code:
NGPUS=1 # number of GPUS (same as what you requested)
MIN_SIGNAL=1.0e-5
CHUNK_SIZE=90000 # GPU paralleled file writing size
ANGLE_DEG=60 # search angle
MAX_ANGLE=1.04719753 # ANGLE_DEG*pi/180
THRESHOLD=0.1
STEP_SIZE=0.5
SEED_DENS=1

# # From Wynton:
gpustreamlines_path=/protected/data/rajlab1/shared_data/singularity/images/gpustreamlines.sif
# # From Radiology SCS:
# gpustreamlines_path=/data/i2/software/singularity/gpustreamlines.sif


#Make Directories for Outputs
mkdir $bids_root_dir/derivatives/pipetography/sub-$1/ses-$2/tracktography
mkdir -p $bids_root_dir/derivatives/streamlines/sub-$1/ses-$2

# Inputs:
#     - BVEC={PATH TO}.bvec
#       BVAL={PATH TO}.bval
#       DTI={PATH TO}.nii.gz
#       MASK=Streamline possible locations mask - whole brain mask
#       ROI=Seed start mask - grey matter white matter interface
BVEC='/input/dwi_space-acpc_res-1mm.bvecs'
BVAL='/input/dwi_space-acpc_res-1mm.bvals'
DTI='/input/dwi_space-acpc_res-1mm.nii.gz'
MASK='/input/dwi_space-acpc_res-1mm_seg-brain_mask.nii.gz'
#ROI='/input/T1w_space-acpc_seg-gmwmi_mask.nii.gz' #Removed for whole-brain seeding
PREFIX='/opt/exec/output/tmp-sub01'  # prefix to name temporary chunk files
TMPFILES='tmp-sub01*' # temp output file with wild card ex: tmp-*

# 2. Execute singularity container:
singularity exec --nv -B $bids_root_dir/derivatives/pipetography/sub-$1/ses-$2/tracktography:/opt/exec/output -B $bids_root_dir/derivatives/pipetography/sub-$1/ses-$2/preprocessed:/input $gpustreamlines_path python /opt/exec/run_dipy_gpu.py $DTI $BVAL $BVEC $MASK --roi_nifti $MASK --chunk-size $CHUNK_SIZE --output-prefix $PREFIX --ngpus $NGPUS --use-fast-write --max-angle $MAX_ANGLE --min-signal $MIN_SIGNAL --tc-threshold $THRESHOLD --step-size $STEP_SIZE --sampling-density $SEED_DENS

echo "Merging track chunks..."
# 3. Merge chunks back together to 1 trk file
singularity exec --nv --pwd /opt/exec/output -B $bids_root_dir/derivatives/pipetography/sub-$1/ses-$2/tracktography:/opt/exec/output $gpustreamlines_path /opt/exec/merge_trk.sh $TMPFILES -o "sub-$1_ses-$2_gmwmi2wm.trk" 

echo "Deleting track chuncks..."
# 5. Delete all tmp chunks 
cd $bids_root_dir/derivatives/pipetography/sub-$1/ses-$2/tracktography 
rm $TMPFILES

echo "Converting TRK file to TCK..."
# 6. Convert trk to tck for mrtrix3 compatibility
$bids_root_dir/code/trk2tck -i $bids_root_dir/derivatives/pipetography/sub-$1/ses-$2/tracktography/"sub-$1_ses-$2_gmwmi2wm.trk" -o $bids_root_dir/derivatives/pipetography/sub-$1/ses-$2/tracktography/"sub-$1_ses-$2_gmwmi2wm.tck" -c 3

mv $bids_root_dir/derivatives/pipetography/sub-$1/ses-$2/tracktography/"sub-$1_ses-$2_gmwmi2wm.tck" $bids_root_dir/derivatives/streamlines/sub-$1/ses-$2
rm -r $bids_root_dir/derivatives/pipetography/sub-$1/ses-$2/tracktography

echo "Streamline generation for sub-${1} ses-${2} complete."


# # # # # # # # # # Connectome Job # # # # # # # # # # # # # # # # # # # 

echo "Initiating singularity for connectome job..."

singularity exec -e -B $bids_root_dir:/BIDS_dir --pwd $PWD $pipetography_path python /BIDS_dir/code/connectome_def_singlesub.py $1 $2

date
echo "Connectome Job script for sub-{$1} complete."
#Approximate Run Time per subject = 50 minutes

echo "Pipetography Processing Complete."

# # Job Summary Wynton:
qstat -j $JOB_ID

# # Job Summary Radiology SCS:
# squeue -j $SLURM_JOB_ID -o "%.18i %.9P %.8j %.8c %.8u %.8T %.10M %.9l %.6D %R %D"

connectome_def_singlesub.py
import os
import json
import bids
from itertools import product
from glob import glob
import pipetography.connectomes as pc
from sys import argv
import re

sub = int(argv[1].strip()) #idx is [1] because [0] is the script name
ses = str(argv[2].strip())

def main():
	bids_folder = os.path.abspath('..')
	bids_layout = bids.layout.BIDSLayout(bids_folder)
	sub_list = bids_layout.get_subjects()
	ses_list = bids_layout.get_sessions()
	all_sub_ses_combos = set(product(sub_list, ses_list))

	skip_these = []
	for tup in all_sub_ses_combos:
		sub_ses = os.path.join(bids_folder, 'sub-'+tup[0], 'ses-'+tup[1], 'dwi')
		if not os.path.exists(sub_ses):
			skip_these.append(tup)
		elif int(tup[0])!=sub or str(tup[1])!=ses:
			skip_these.append(tup)
	

	print('Subject-Session tuples to process:')
	print(all_sub_ses_combos.difference(skip_these))


	connectomes = pc.connectome(
		BIDS_dir = '/BIDS_dir',
		skip_tuples = skip_these,
		debug=False,
		atlas_list = ['/BIDS_dir/code/Pipetography_Atlases/desikanKilliany86MNI.nii.gz','/BIDS_dir/code/Pipetography_Atlases/BN_Atlas_246_1mm.nii.gz','/BIDS_dir/code/Pipetography_Atlases/aal116MNI.nii.gz']
		)
    # connectomes.subject_template
	connectomes.create_nodes()
	connectomes.connect_nodes(wf_name='connectomes') #connectomes has to be the working folder name
	connectomes.run_pipeline(parallel=3)
    
if __name__ == '__main__':
    main()


connectome_job_singlesub.sh
#!/bin/bash

# # Wynton Submissions:
#$ -S /bin/bash
#$ -N PP_Connect
#$ -cwd
#$ -j y
#$ -l mem_free=40G
#$ -l h_rt=12:00:00 #note: time limit is 7-days
#$ -o /wynton/protected/home/rad-brain/bsipes/outlog_general

# # Radiology SCS Submissions:
# #SBATCH --job-name=PP_Connect
# #SBATCH --nodes=1
# #SBATCH --ntasks=1
# #SBATCH --cpus-per-task=10
# #SBATCH --mem=40G
# #SBATCH --time=12:00:00 #note: time limit is 2-days
# #SBATCH --output=/home/%u/slurm/%x-%j.log

/bin/echo Running on host: `hostname`.
/bin/echo In directory: `pwd`
/bin/echo Starting on: `date`


bids_root_dir=`pwd`/..

# # From Wynton:
pipetography_path=/protected/data/rajlab1/shared_data/singularity/images/pipetography_latest.sif
# # From Radiology SCS:
# pipetography_path=/data/i2/software/singularity/pipetography_latest.sif

# Singularity container call:
echo "Initiating singularity pipetography connectome job for sub-${1} ses-${2}..."

singularity exec -e -B $bids_root_dir:/BIDS_dir --pwd $PWD $pipetography_path python /BIDS_dir/code/connectome_def_singlesub.py $1 $2

date
echo "Connectome Job script for sub-${1} complete."
#Approximate Run Time per subject = 50 minutes

# # Job Summary Wynton:
qstat -j $JOB_ID

# # Job Summary Radiology SCS:
# squeue -j $SLURM_JOB_ID -o "%.18i %.9P %.8j %.8c %.8u %.8T %.10M %.9l %.6D %R %D"


preprocess_def_singlesub.py
import os
import json
import bids
from itertools import product
from glob import glob
import pipetography.pipeline as pp
from sys import argv
import re

sub = int(argv[1].strip()) #idx is [1] because [0] is the script name
ses = str(argv[2].strip())

def main():
	bids_folder = os.path.abspath('..')
	bids_layout = bids.layout.BIDSLayout(bids_folder)
	sub_list = bids_layout.get_subjects()
	ses_list = bids_layout.get_sessions()
	all_sub_ses_combos = set(product(sub_list, ses_list))

	skip_these = []
	for tup in all_sub_ses_combos:
		sub_ses = os.path.join(bids_folder, 'sub-'+tup[0], 'ses-'+tup[1], 'dwi')
		if not os.path.exists(sub_ses):
			skip_these.append(tup)
		elif int(tup[0])!=sub or str(tup[1])!=ses:
			skip_these.append(tup)

	
	print('Subject-Session tuples to process:')
	print(all_sub_ses_combos.difference(skip_these))

	#starting pipeline
	dwi_preproc = pp.pipeline(
		BIDS_dir='/BIDS_dir',
		ext='nii.gz',
		rpe_design='-rpe_none',
		regrid=True,
		gmwmi=False,
		mrtrix_nthreads=8,
		skip_tuples=skip_these,
		debug=False)
	dwi_preproc.create_nodes()
	dwi_preproc.connect_nodes(rpe_design='-rpe_none')
	dwi_preproc.workflow.base_dir = '/BIDS_dir/code/workingdir'
	dwi_preproc.run_pipeline(parallel=2)

if __name__ == '__main__':
	main()


preprocess_job_singlesub.sh
#!/bin/bash

# # Wynton Submissions:
#$ -S /bin/bash
#$ -N PP_Preproc
#$ -cwd
#$ -j y
#$ -l mem_free=40G
#$ -l h_rt=12:00:00 #note: time limit is 7-days
#$ -o /wynton/protected/home/rad-brain/bsipes/outlog_general

# # Radiology SCS Submissions:
# #SBATCH --job-name=PP_Preproc
# #SBATCH --nodes=1
# #SBATCH --ntasks=1
# #SBATCH --cpus-per-task=10
# #SBATCH --mem=40G
# #SBATCH --time=12:00:00 #note: time limit is 2-days
# #SBATCH --output=/home/%u/slurm/%x-%j.log

/bin/echo Running on host: `hostname`.
/bin/echo In directory: `pwd`
/bin/echo Starting on: `date`


bids_root_dir=`pwd`/..

# # From Wynton:
pipetography_path=/protected/data/rajlab1/shared_data/singularity/images/pipetography_latest.sif
# # From Radiology SCS:
# pipetography_path=/data/i2/software/singularity/pipetography_latest.sif

# Singularity container call:
echo "Initiating singularity pipetography preprocesing job for sub-${1} ses-${2}..."

singularity exec -e -B $bids_root_dir:/BIDS_dir --pwd $PWD $pipetography_path python /BIDS_dir/code/preprocess_def_singlesub.py $1 $2

date
echo "Preprocessing Job script for sub-${1} ses-${2} complete."
#Approximate Run Time per subject = 1-2 hours

# # Job Summary Wynton:
qstat -j $JOB_ID

# # Job Summary Radiology SCS:
# squeue -j $SLURM_JOB_ID -o "%.18i %.9P %.8j %.8c %.8u %.8T %.10M %.9l %.6D %R %D"


pipetography_all_HCP.sh
#!/bin/bash

# # Wynton Submissions:
#$ -S /bin/bash
#$ -N PP_All
#$ -cwd
#$ -q gpu.q
#$ -j y
#$ -l mem_free=64G
#$ -l h_rt=24:00:00 #gross overestimate, probably closer to 5-10 hours
#$ -o /wynton/protected/home/rad-brain/bsipes/outlog_general

# # Radiology SCS Submissions:
# #SBATCH --job-name=PP_All
# #SBATCH --nodes=1
# #SBATCH --ntasks=1
# #SBATCH --nodelist=cronus #I've had best luck with cronus, although other nodes may work fine, too.
# #SBATCH -N 1 --partition=gpu --ntasks-per-node=1
# #SBATCH --gres=gpu:1
# #SBATCH --mem=64G
# #SBATCH --time=24:00:00 #gross overestimate, probably closer to 5-10 hours
# #SBATCH --output=/home/%u/slurm/%x-%j.log

/bin/echo Running on host: `hostname`.
/bin/echo In directory: `pwd`
/bin/echo Starting on: `date`


bids_root_dir=`pwd`/..

# # From Wynton:
pipetography_path=/protected/data/rajlab1/shared_data/singularity/images/pipetography_latest.sif
# # From Radiology SCS:
# pipetography_path=/data/i2/software/singularity/pipetography_latest.sif


# # # # # # # # # # Preprocessing Job # # # # # # # # # # # # # # # # # # # 

echo "Initiating singularity for preprocesing job..."

singularity exec -e -B $bids_root_dir:/BIDS_dir --pwd $PWD $pipetography_path python /BIDS_dir/code/preprocess_def_singlesub.py $1 $2

date
echo "Preprocessing Job script for sub-{$1} complete."

# # # # # # # # # # Streamlines Job # # # # # # # # # # # # # # # # # # # 


echo "Running Tractography for sub-${1} ses-${2}"

# Default parameters for the streamline generation code:
NGPUS=1 # number of GPUS (same as what you requested)
MIN_SIGNAL=1.0e-5
CHUNK_SIZE=90000 # GPU paralleled file writing size
ANGLE_DEG=60 # search angle
MAX_ANGLE=1.04719753 # ANGLE_DEG*pi/180
THRESHOLD=0.1
STEP_SIZE=0.5
SEED_DENS=1

# # From Wynton:
gpustreamlines_path=/protected/data/rajlab1/shared_data/singularity/images/gpustreamlines.sif
# # From Radiology SCS:
# gpustreamlines_path=/data/rajlab1/shared_data/singularity/images/gpustreamlines.sif


#Make Directories for Outputs
mkdir $bids_root_dir/derivatives/pipetography/sub-$1/ses-$2/tracktography
mkdir -p $bids_root_dir/derivatives/streamlines/sub-$1/ses-$2

# Inputs:
#     - BVEC={PATH TO}.bvec
#       BVAL={PATH TO}.bval
#       DTI={PATH TO}.nii.gz
#       MASK=Streamline possible locations mask - whole brain mask
#       ROI=Seed start mask - grey matter white matter interface
BVEC='/input/dwi_space-acpc_res-1mm.bvecs'
BVAL='/input/dwi_space-acpc_res-1mm.bvals'
DTI='/input/dwi_space-acpc_res-1mm.nii.gz'
MASK='/input/dwi_space-acpc_res-1mm_seg-brain_mask.nii.gz'
#ROI='/input/T1w_space-acpc_seg-gmwmi_mask.nii.gz' #Removed for whole-brain seeding
PREFIX='/opt/exec/output/tmp-sub01'  # prefix to name temporary chunk files
TMPFILES='tmp-sub01*' # temp output file with wild card ex: tmp-*

# 2. Execute singularity container:
singularity exec --nv -B $bids_root_dir/derivatives/pipetography/sub-$1/ses-$2/tracktography:/opt/exec/output -B $bids_root_dir/derivatives/pipetography/sub-$1/ses-$2/preprocessed:/input $gpustreamlines_path python /opt/exec/run_dipy_gpu.py $DTI $BVAL $BVEC $MASK --roi_nifti $MASK --chunk-size $CHUNK_SIZE --output-prefix $PREFIX --ngpus $NGPUS --use-fast-write --max-angle $MAX_ANGLE --min-signal $MIN_SIGNAL --tc-threshold $THRESHOLD --step-size $STEP_SIZE --sampling-density $SEED_DENS

echo "Merging track chunks..."
# 3. Merge chunks back together to 1 trk file
singularity exec --nv --pwd /opt/exec/output -B $bids_root_dir/derivatives/pipetography/sub-$1/ses-$2/tracktography:/opt/exec/output $gpustreamlines_path /opt/exec/merge_trk.sh $TMPFILES -o "sub-$1_ses-$2_gmwmi2wm.trk" 

echo "Deleting track chuncks..."
# 5. Delete all tmp chunks 
cd $bids_root_dir/derivatives/pipetography/sub-$1/ses-$2/tracktography 
rm $TMPFILES

echo "Converting TRK file to TCK..."
# 6. Convert trk to tck for mrtrix3 compatibility
$bids_root_dir/code/trk2tck -i $bids_root_dir/derivatives/pipetography/sub-$1/ses-$2/tracktography/"sub-$1_ses-$2_gmwmi2wm.trk" -o $bids_root_dir/derivatives/pipetography/sub-$1/ses-$2/tracktography/"sub-$1_ses-$2_gmwmi2wm.tck" -c 3

mv $bids_root_dir/derivatives/pipetography/sub-$1/ses-$2/tracktography/"sub-$1_ses-$2_gmwmi2wm.tck" $bids_root_dir/derivatives/streamlines/sub-$1/ses-$2
rm -r $bids_root_dir/derivatives/pipetography/sub-$1/ses-$2/tracktography

echo "Streamline generation for sub-${1} ses-${2} complete."


# # # # # # # # # # Connectome Job # # # # # # # # # # # # # # # # # # # 

echo "Initiating singularity for connectome job..."

singularity exec -e -B $bids_root_dir:/BIDS_dir --pwd $PWD $pipetography_path python /BIDS_dir/code/connectome_def_singlesub.py $1 $2

date
echo "Connectome Job script for sub-{$1} complete."
#Approximate Run Time per subject = 50 minutes

echo "Pipetography Processing Complete."

# # Job Summary Wynton:
qstat -j $JOB_ID

# # Job Summary Radiology SCS:
# squeue -j $SLURM_JOB_ID -o "%.18i %.9P %.8j %.8c %.8u %.8T %.10M %.9l %.6D %R %D"